### PR TITLE
Adds some check helpers for validation of field values

### PR DESCRIPTION
**Pull request by bowei**

### CODE REVIEW

File: pkg/cloud/api/check.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+				if !fv.IsZero() {
+					return false, fmt.Errorf("%s has a non-zero value (%v) but is a System field", fv.Interface(), fp)
+				}
+			case FieldTypeOutputOnly:
+				if !fv.IsZero() {
+					return false, fmt.Errorf("%s has a non-zero value (%v) but is an OutputOnly field", fv.Interface(), fp)
+				}
+			case FieldTypeOrdinary:
+				switch {
+				case fv.IsZero() && !(inNull(ft.Name) || inForceSend(ft.Name)):
```
Comment (briantkennedy) [inline pkg/cloud/api/check.go:72]: nit: `!inNull(ft.Name) && !inForceSend(ft.Name)` is easier to read than `!(inNull(ft.Name) || inForceSend(ft.Name))` (I had to visualize the truth table in my head then invert).

File: pkg/cloud/api/check.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		for i := 0; i < v.NumField(); i++ {
+			ft := v.Type().Field(i)
+			if ft.Name == "NullFields" || ft.Name == "ForceSendFields" {
+				continue
+			}
+			fType := traits.fieldType(p.Field(ft.Name))
+			fv := v.Field(i)
+			fp := p.Field(ft.Name)
+
+			switch fType {
```
Comment (briantkennedy) [inline pkg/cloud/api/check.go:61]: does this need a `default:` case?

**Review by briantkennedy — APPROVED** [review]

File: pkg/cloud/api/check.go (hunk 1 above)
Comment (bowei) [reply pkg/cloud/api/check.go:72]: Good point -- thanks for running the Karnaugh map optimization for me

**Review by briantkennedy — APPROVED** [review]

**Pull request merged**